source/parser.hera

```diff
@@ -3565,7 +3590,10 @@ JSXChild
 # https://facebook.github.io/jsx/#prod-JSXText
 JSXText
   # NOTE: not currently excluding https://facebook.github.io/jsx/#prod-HTMLCharacterReference
-  [^{}<>]+
+  # NOTE: Additionally forbidding leading whitespace, so it can be used for
+  # indentation (by JSXNestedChildren) or ignored by JSXChildren,
+  # and newlines which we leave for the next indentation.
+  [^{}<>\s][^{}<>\n\r]*
```

Comment:
This might not be necessary since the `Nested JSXChild` rule should consume and hold the space before getting to here.

Reply:
I had a ton of extremely difficult to debug bugs before adding this; feel free to try, but I'm pretty sure it's necessary. The issue is that `<foo>` can also be parsed as `< foo >` given certain stuff on the preceding and following lines.

Reply:
It looks like stopping at EOL was the key ingredient: https://github.com/DanielXMoore/Civet/commit/fe92ceda94883f4e35c638ed3ce5d9a376db7e2b